persistence.js

```diff
@@ -309,23 +280,25 @@ RedisPersistence.prototype._setup = function () {
 
   var splitStream = through.obj(split)
 
-  var hgetallStream = throughv.obj(function getStream (chunk, enc, cb) {
-    var pipeline = that._getPipeline()
-    pipeline.hgetallBuffer(chunk, cb)
+  var hgetallStream = throughv.obj(function getStream (clientId, enc, cb) {
+    var clientSubKey = clientKey + clientId
+    that._db.hgetall(clientSubKey, function clientHash (err, hash) {
+      cb(err, {clientHash: hash, clientId: clientId})
+    })
   }, function emitReady (cb) {
     that.ready = true
     that.emit('ready')
     cb()
   })
-  .on('data', function processKeys (all) {
-    processKeysForClient(all, that)
+  .on('data', function processKeys (data) {
+    processKeysForClient(data.clientId, data.clientHash, that)
   })
 
-  this._db.lrange(subscriptionsKey, 0, 10000, function lrangeResult (err, results) {
+  this._db.smembers(clientsKey, function smembers (err, clientIds) {
     if (err) {
       splitStream.emit('error', err)
     } else {
-      splitStream.write(results)
+      splitStream.write(clientIds)
```

Comment:
You can completely avoid the use of `splitStream` here, just do the processing and call write on the `throughv` instance